docs/extensions/tutorials/_posts/1970-01-01-SettingLocalEnvironment.md

```diff
@@ -22,75 +19,67 @@ Before setting up a local environment for building Shoutem extensions, set up Re
 
-```ShellSession
-$ shoutem run-ios
-Select a device: iPhone 6
-Select your app: Restaurants ({{ site.example.appId }})
-Running `Restaurants` app on `iPhone 6` simulator...
-...
-```
+#### Shoutem Run
 
-This command takes the code from Shoutem Builder. For Android, use `shoutem run-android`.
+`shoutem run` is used to preview the app on your device using the the **Shoutem Preview app**. The great thing about the Shoutem Preview app is that it will let you see what the app looks like on iOS even if you're using Windows.
 
-## Local Development
+However, there are limitations to what the Shoutem Preview app can preview. Namely, it can only preview code that has no native code linked into it. This is because the Shoutem Preview app has it's own binary, so it can only preview changes made to the JavaScript bundle of the app.
 
-_Pushing_ your extension to Shoutem and waiting for Shoutem to build the new app is time consuming. Just running your app locally doesn't solve this problem since it still takes the code from Shoutem.
+#### Shoutem Run-OS
 
-However, you can _link_ your local extension code to the local environment (similar to [react-native link](https://facebook.github.io/react-native/docs/linking-libraries-ios.html)). Next time you run the app, Shoutem will take the local extension code instead of taking it from the Builder, so you don't have to wait through the `push` and download.
+`shoutem run-ios` and `shoutem run-android` are used for previewing the app locally on an emulator. They work analogously to `react-native run-ios` and `react-native run-android`, so they require you to set up your React Native environment as stated earlier in the tutorial.
 
-Navigate to your extension directory:
+#### Shoutem Builder
 
-```ShellSession
-$ cd restaurants
-```
+You can of course preview your app through the Shoutem Builder where you can manage your app as well. It works identically to the Shoutem Preview app, having the same advantages and limitations.
 
-... and link the extension:
+## Local Development
+
+We will now explain how to preview code changes in your extensions in real-time. If you've gone through the **Getting Started** tutorial, you should have an app on the Builder. To be able to see changes in your extension as you make them **without** having to push your extension to Shoutem every time you make a change, you'll have to _link_ your local extension code to the local environment (similar to [react-native link](https://facebook.github.io/react-native/docs/linking-libraries-ios.html)).
+
+Navigate to your extension directory and link it to your local environment:
 
 ```ShellSession
+$ cd hello-world
 $ shoutem link
-Extension successfully linked. Please, kill the packager before running the app.
+Directory successfully linked. Please, kill the packager before running the app.
```

Comment:
It should be "Extension". Directory is a bug on CLI, but nevermind, we'll anyways change that soon.